contracts/connection-router/tests/test.rs

```diff
@@ -62,18 +57,26 @@ fn make_chain(name: &str, config: &mut TestConfig) -> Chain {
 }
 
 fn register_chain(config: &mut TestConfig, chain: &Chain) {
-    let _ = config
-        .app
-        .execute_contract(
-            config.governance_address.clone(),
-            config.contract_address.clone(),
-            &ExecuteMsg::RegisterChain {
-                chain: chain.chain_name.clone(),
-                gateway_address: chain.gateway.to_string(),
-            },
-            &[],
-        )
-        .unwrap();
+    let _ = config.connection_router.execute(
+        &mut config.app,
+        config.governance_address.clone(),
+        &ExecuteMsg::RegisterChain {
+            chain: chain.chain_name.clone(),
+            gateway_address: chain.gateway.to_string(),
+        },
+    );
+    // let _ = config
+    //     .app
+    //     .execute_contract(
+    //         config.governance_address.clone(),
+    //         config.contract_address.clone(),
+    //         &ExecuteMsg::RegisterChain {
+    //             chain: chain.chain_name.clone(),
+    //             gateway_address: chain.gateway.to_string(),
+    //         },
+    //         &[],
+    //     )
+    //     .unwrap();
 }
 
 fn generate_messages(
```

Comment:
not in the scope of this PR, I just wanted to point it out to you: I think this function would be much cleaner like this:
```
fn generate_messages(
    src_chain: &Chain,
    dest_chain: &Chain,
    nonce: usize,
    count: usize,
) -> Vec<Message> {
    (nonce..(nonce+count))
    .map(|nonce| 
        Message {
            cc_id: CrossChainId {
                id: format!("tx_id:{}", nonce).parse().unwrap(),
                chain: src_chain.chain_name.clone(),
            },
            destination_address: "idc".parse().unwrap(),
            destination_chain: dest_chain.chain_name.clone(),
            source_address: "idc".parse().unwrap(),
            payload_hash: [nonce as u8; 32],
        }
    ).collect()
}
```
that way you don't leak any mutable state and you don't have to worry about the details of the vector creation

Reply:
Thanks, updated!